src/Form/LinkBlockFormDataProvider.php

```diff
@@ -0,0 +1,272 @@
+    {
+        $this->idLinkBlock = $idLinkBlock;
+
+        return $this;
+    }
+
+    /**
+     * @param array $data
+     * @return array
+     */
+    private function validateLinkBlock(array $data)
+    {
+        $errors = [];
+        if (!isset($data['id_hook'])) {
+            $errors[] = [
+                'key' => "Missing id_hook",
+                'domain' => 'Admin.Catalog.Notification',
+                'parameters' => [],
+            ];
+        }
+
+        if (!isset($data['block_name'])) {
+            $errors[] = [
+                'key' => "Missing block_name",
+                'domain' => 'Admin.Catalog.Notification',
+                'parameters' => [],
+            ];
+        } else {
+            foreach ($this->languages as $language) {
+                if (
+                    !isset($data['block_name'][$language['id_lang']]) ||
+                    empty($data['block_name'][$language['id_lang']])
+                ) {
+                    $errors[] = [
+                        'key' => "Missing block_name value for language %s",
+                        'domain' => 'Admin.Catalog.Notification',
+                        'parameters' => [$language['iso_code']],
+                    ];
+                }
+            }
+        }
+
+        if (isset($data['custom'])) {
+            foreach ($data['custom'] as $customIndex => $custom) {
+                foreach ($this->languages as $language) {
+                    if (!isset($custom[$language['id_lang']])) {
+                        $errors[] = [
+                            'key' => "Missing block_name value for language %s",
+                            'domain' => 'Admin.Catalog.Notification',
+                            'parameters' => [$language['iso_code']],
+                        ];
+                    } else {
+                        $langCustom = $custom[$language['id_lang']];
+                        $fields = ['title', 'url'];
+                        foreach ($fields as $field) {
+                            if (!isset($langCustom[$field]) || empty($langCustom[$field])) {
+                                $errors[] = [
+                                    'key' => "Missing %s value in custom[%s] for language %s",
+                                    'domain' => 'Admin.Catalog.Notification',
+                                    'parameters' => [$field, $customIndex, $language['iso_code']],
+                                ];
+                            }
+                        }
+                    }
+                }
+            }
+        }
+
+
+        return $errors;
+    }
+
+    /**
+     * Register the selected hook to this module if it was not registered yet
+     *
+     * @param int $hookId
+     *
+     * @throws \PrestaShopException
+     */
+    private function updateHook($hookId)
+    {
+        $hookName = \Hook::getNameById($hookId);
+        $module = \Module::getInstanceByName('ps_linklist');
```

Comment:
PrestaShop\PrestaShop\Core\Addon\Module\ModuleRepository->getInstanceByName is maybe better?

Reply:
oh great I didn't know this one! thanks